tools/fuchsia/with_envs.py

```diff
@@ -0,0 +1,52 @@
+#!/usr/bin/env python3
+# Copyright (c) 2024, the Flutter project authors. All rights reserved.
+# Use of this source code is governed by a BSD-style license that can be found
+# in the LICENSE file.
+
+import os
+import platform
+import subprocess
+import sys
+
+sys.path.insert(
+    0,
+    os.path.abspath(
+        os.path.join(
+            os.path.dirname(__file__), '../../../fuchsia/test_scripts/test/'
+        )
+    )
+)
+
+from common import catch_sigterm, wait_for_sigterm
+
+
+def Main():
+  """
+    Executes the test-scripts with required environment variables. It acts like
+    /usr/bin/env, but provides some extra functionality to dynamically set up
+    the environment variables.
+    """
+  # Ensures the signals can be correctly forwarded to the subprocesses.
+  catch_sigterm()
+
+  # Flutter uses a different repo structure and fuchsia sdk is not in the
+  # third_party/, so setting the DIR_SRC_ROOT does not make sense.
+  os.environ['FUCHSIA_IMAGES_ROOT'] = os.path.abspath(
+      os.path.join(os.path.dirname(__file__), '../../../fuchsia/images/')
+  )
+
+  assert platform.system() == 'Linux', 'Unsupported OS ' + platform.system()
+  os.environ['FUCHSIA_SDK_ROOT'] = os.path.abspath(
+      os.path.join(os.path.dirname(__file__), '../../../fuchsia/sdk/linux')
+  )
+
+  with subprocess.Popen(sys.argv[1:]) as proc:
```

Comment:
Is this consuming the exit code? It looks like from https://logs.chromium.org/logs/flutter/buildbucket/cr-buildbucket/8759257676482292513/+/u/test:_x64_emulator_based_tests/stdout we would expect the return code to be 1, but the test is marked as green.

Reply:
I was about asking the question. Yes, it looks like the problem.

I can add a TODO to not consuming the exit code, or the CI will fail with the linux_fuchsia.json change due to the lack of kvm.

Reply:
I'm not sure there's open source docs yet, but I found go/flutter-android-emulators

tl;dr is we can edit https://github.com/flutter/engine/blob/main/.ci.yaml to add

```
dimensions:
    kvm: “1”,
    cores: “8”,
    Machine_name: “n1-standard-8”
```

to the `Linux linux_fuchsia` target which will run it on machines that have KVM.

@ricardoamador can you confirm, or is there anything different we need to do for engine_v2?

Reply:
That should be enough to pick the machine with kvm. Though I have seen that there are other types of machines with kvm that are not n1-stardard-8. Not sure if that is a requirement here.

Reply:
I added kvm: "1", # of cores and machine type may not be important.

But does the .ci.yaml change impact the current ci run?

Reply:
Yes, .ci.yaml will take effect in presubmit. I'm interested to know if this applies to the drones spawned by the ci/builders code (engine_v2), or if we need to add the dimension there as well.

Reply:
Your changes should take effect unless you are changing the target name, etc... in which case you would need to wait for the ci.yaml roller.

Reply:
@CaseyHillers This should also work for engine_v2 as well since the configs are in the config yaml file specified in the "config_name" property and not in the top level ci.yaml file.